### PR TITLE
Updates for ots 9.1.0

### PR DESCRIPTION
**Pull request by josh-hadley**
[config] Use ots v9.1.0 & updated sha256
[setup] set ECA std (back) to c++11
[py] remove commented-out code
[tests] clarify wording in test assertion message
[cpp] use int instead of bool for "quiet" param (alignment issue on arm64 was causing segfault) 
[ci] drop Python 3.7, add Python 3.11 builds
[setup] project classifiers: drop Python 3.7, add Python 3.11 [README] drop Python3.7, add Python 3.11 (requirements section)
[ci] don't use set-output
[ci] use arm64 (not universal2), fix release notes string, add arch flags for CIBW

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -41,8 +41,9 @@ jobs:
       with:
         output-dir: dist
       env:
-        CIBW_BUILD: "cp37-* cp38-* cp39-* cp310-*"
-        CIBW_ARCHS_MACOS: x86_64 universal2
+        CIBW_BUILD: "cp38-* cp39-* cp310-* cp311-*"
+        CIBW_ARCHS_MACOS: x86_64 arm64
+        CIBW_ENVIRONMENT_MACOS: "CFLAGS='-arch arm64 -arch x86_64' CXXFLAGS='-arch arm64 -arch x86_64' LDFLAGS='-arch arm64 -arch x86_64'"
```
Comment (anthrotype) [inline .github/workflows/release.yml:46]: why did you remove `universal2` from `CIBW_ARCHS_MACOS` at the same time as `CIBW_ENVIRONMENT_MACOS` contains both architecture flags `-arch arm64 -arch x86_64`?

Comment (josh-hadley) [reply .github/workflows/release.yml:46]: > why did you remove universal2 from CIBW_ARCHS_MACOS

Initially just to diagnose why the `arm64` part of the `universal2` wheel was broken (it has been since the initial introduction of that wheel, but until recently was something I was unable to check).

> CIBW_ENVIRONMENT_MACOS contains both architecture flags -arch arm64 -arch x86_64

Not having `-arch arm64` C/CXX/LDFLAGS set resulted in the `arm64` binary building incorrectly. So you could install the wheel, but when you went to use `pyots` it would not function correctly. Turns out the ENVIRONMENT flags do not control what binary gets built (the `CIBW_ARCHS_MACOS` and related settings do that), but the flags need to be set when the architecture in question is being built (for `arm64` at least). If you check the resulting binaries (.so files in the package) with e.g. `file` or `lipo` you can see that the `x86_64` version contains only `x86_64` and the `arm64` one contains only `arm64`. For CIBW there's only one opportunity to set the ENVIRONMENT flags, so I set both because I'm building both wheels, and everything now works as expected.

I was initially planning to reset to `universal2` once I got the `arm64` situation figured out, but after some thought that seemed wasteful (the wheel is ~2x the size of `x86_64` or `arm64` individual builds) so I just left it as independent. Do you have a need for the `universal2` wheel?

File: setup.py [hunk 2]
```diff
@@ -279,10 +279,10 @@ def run(self):
     'Intended Audience :: Developers',
     'Topic :: Software Development :: Testing',
     'License :: OSI Approved :: BSD License',
-    'Programming Language :: Python :: 3.7',
     'Programming Language :: Python :: 3.8',
     'Programming Language :: Python :: 3.9',
     'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
```
Comment (miguelsousa) [inline setup.py:285]: `python_requires` item further down needs bumping

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -96,6 +97,6 @@ jobs:
       env:
```
Comment (miguelsousa) [inline .github/workflows/release.yml:97]: You can modernize this workflow even further by,
1. removing this `env` parameter
2. adding these two lines right below the `runs-on` parameter of the `publish_release` job
```yaml
    permissions:
      contents: write  # Required by 'action-gh-release'
```

**Review by miguelsousa — APPROVED** [review]
1 required change + 1 suggested change. Otherwise LGTM

File: setup.py (hunk 2 above)
Comment (josh-hadley) [reply setup.py:285]: Done in https://github.com/adobe-type-tools/pyots/pull/25/commits/b43a091a21bea491603721609a91fea9882688d4

File: .github/workflows/release.yml (hunk 3 above)
Comment (josh-hadley) [reply .github/workflows/release.yml]: Done in https://github.com/adobe-type-tools/pyots/pull/25/commits/108f86650de3195e580b5bbd7fc23c9522da7f01
